src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-user-details/rgw-user-details.component.ts

```diff
@@ -59,8 +59,8 @@ export class RgwUserDetailsComponent implements OnChanges, OnInit {
   }
 
   ngOnChanges() {
-    if (this.selection.hasSelection) {
-      this.user = this.selection.first();
+    if (typeof this.selection !== 'undefined') {
+      this.user = this.selection;
```

Comment:
```suggestion
```

Reply:
https://tracker.ceph.com/issues/44410
I opened a issue to track your refactor request. I changed the ones with just a small diff, but this would cause too many changes in my opinion. It would get even harder to review this pr.